lighthouse-cli/cli-flags.js

```diff
@@ -12,19 +12,32 @@ const pkg = require('../package.json');
 const printer = require('./printer');
 
 /**
- * @param {string=} manualArgv
+ * @param {(string | string[])[]} arr
+ * @return {string[]}
+ */
+function flatten(arr) {
+  /** @type {string[]} */
+  const result = [];
+  return result.concat(...arr);
+}
+
+/**
+ * @param {string[]} manualArgv
  * @return {LH.CliFlags}
  */
-function getFlags(manualArgv) {
-  // @ts-ignore yargs() is incorrectly typed as not returning itself
-  const y = manualArgv ? yargs(manualArgv) : yargs;
-  return y.help('help')
+function getFlags(...manualArgv) {
+  const y = manualArgv.length ? yargs(manualArgv) : yargs;
+  /** @type {LH.CliFlags} */
```

Comment:
this doesn't buy us much except having to coerce with `/** @type {string[]} */` below

Reply:
would like to keep `/** @type {string[]} */` otherwise more errors come up. I added a comment saying why we can assume it to be a string[].

`LH.CliFlags` one could be removed if you want.

Reply:
> would like to keep /** @type {string[]} */ otherwise more errors come up.

but it's wrong because *all* of this is wrong :)

The ts-ignore was suppressing a different error than it said (I assume/hope it changed at some point), but the comment was still correct ('`yargs()` is incorrectly typed as not returning itself'): all the calls on yargs after `help()` are of type `any` and return `any`. In addition:
- the type of `LH.CliFlags[keyof LH.CliFlags]` is definitely *not* a string array, really it should be `LH.CliFlags['onlyAudits'|'onlyCategories'|'output'|'plugins'|'skipAudits']`
- if we pretend the type is `string[]` because yargs is returning that for the subset of keys we care about, assigning back as `string[]` is still not correct even for that subset because e.g. `output` is `LH.OutputMode[]`, not `string[]` (and that's ignoring trying to reassign back to `LH.CliFlags[keyof LH.CliFlags]`). This will be an error in tsc 3.5+.

I think the most honest thing is to just leave as `any` internal to `getFlags` with a note about why and leave it up to tests to verify that `getFlags` really is returning  `LH.CliFlags`.